Source/core/JSON.h

```diff
@@ -1934,19 +1934,19 @@ namespace Core {
                 char replacement = current;
                 switch (current) {
                 case 'n':
-                    replacement = '\n';
+                    replacement = '\\n';
```

Comment:
I guess it is not the intent of the code the new value is capped, please see comments in the Pull Request for details.